### PR TITLE
[WIP] Changes to use put_simple et al

### PR DESCRIPTION
**Pull request by mkanoor**
Refactoring MiqQueue.put into different functions

MiqQueue.put_simple
MiqQueue.put_with_guid
....

### CODE REVIEW

**Comment by mkanoor** [comment]
@bzwei 
Please review

File: app/models/authenticator/base.rb [hunk 1]
```diff
@@ -213,7 +213,7 @@ def authorize_queue(username, _request, _options, *args)
       task = MiqTask.create(:name => "#{self.class.proper_name} User Authorization of '#{username}'", :userid => username)
       if authorize_queue?
         encrypt_ldap_password(config) if MiqLdap.using_ldap?
-        MiqQueue.put(
+        MiqQueue.put_with_guid(
```
Comment (kbrock) [inline app/models/authenticator/base.rb:216]: this has status. do you want to punt on that?

File: app/models/blacklisted_event.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def queue_sync_blacklisted_event_names
     _log.info("Queueing sync_blacklisted_event_names for [#{servers.length}] active_miq_servers, ids: #{servers.collect(&:id)}")
 
     servers.each do |s|
-      MiqQueue.put(
+      MiqQueue.put_with_guid(
```
Comment (kbrock) [inline app/models/blacklisted_event.rb:43]: This is here to synchronize configuration.
Wonder if this has a theme too

File: app/models/miq_report/generator/async.rb [hunk 3]
```diff
@@ -54,7 +54,7 @@ def async_generate_table(options = {})
     unless sync # Only queued if sync reporting disabled (default)
       cb = {:class_name => task.class.name, :instance_id => task.id, :method_name => :queue_callback_on_exceptions, :args => ['Finished']}
       unless self.new_record?
-        MiqQueue.put(
+        MiqQueue.put_with_queue(
```
Comment (kbrock) [inline app/models/miq_report/generator/async.rb:57]: does it make sense to get cb into here?
not sure if that is useful in the table

(aah, reuse)

File: app/models/miq_report_result.rb [hunk 4]
```diff
@@ -184,7 +184,10 @@ def async_generate_result(result_type, options = {})
     task.update_status("Queued", "Ok", "Task has been queued")
 
     sync = ::Settings.product.report_sync
-
+    # Queue Pattern, Back ground job
+    # Queue_name should have been reporting
```
Comment (kbrock) [inline app/models/miq_report_result.rb:188]: think we should just make the change

File: app/models/miq_report_result.rb [hunk 5]
```diff
@@ -296,6 +299,8 @@ def self.orphaned_counts_by_userid
   def self.delete_by_userid(userids)
     userids = userids.to_miq_a
     _log.info("Queuing deletion of report results for the following user ids: #{userids.inspect}")
+    # Embedded Conditional in args
```
Comment (kbrock) [inline app/models/miq_report_result.rb:302]: think this is ok.

they are the args for `"destroy_all"`?

Comment (kbrock) [reply app/models/miq_report_result.rb:302]: do we want `put_zone`

File: app/models/zone.rb [hunk 6]
```diff
@@ -203,7 +203,7 @@ def queue_ntp_reload_if_changed
     _log.info("Zone: [#{name}], Queueing ntp_reload for [#{servers.length}] active_miq_servers, ids: #{servers.collect(&:id)}")
 
     servers.each do |s|
-      MiqQueue.put(
+      MiqQueue.put_with_guid(
```
Comment (kbrock) [inline app/models/zone.rb:206]: Think this is `put_os`

**Review by kbrock — COMMENTED** [review]
just need to add the `alias` entries

**Pull request closed without merging**